[#478] Invalid PVC Size

paulocabrita-ionos opened the pull request:
Related to https://github.com/eclipse/packages/issues/478. 

Created a new parameter - storageSize - that allows any user to define the size of the storage created by the Service Device Registry pod. This allows the installation of the Hono in a Cloud Provider that have policies who restrict the size of the storage.

sophokles73 commented on charts/hono/templates/hono-service-device-registry-embedded/hono-service-device-registry-pvc.yaml:
```diff
@@ -24,5 +24,5 @@ spec:
   resources:
     requests:
-      storage: 1Mi
+      {{ .Values.deviceRegistryExample.embeddedJdbcDeviceRegistry.storageSize }}
```
this will need to be
```
storage: {{ .Values.deviceRegistryExample.embeddedJdbcDeviceRegistry.storageSize | quote }}
```

paulocabrita-ionos replied:
Ok.

sophokles73 commented:
You will also need to bump the chart's version in the `Chart.yaml` file so that the CI system correctly recognizes this PR as an actual change. Increasing the patch version by one will do the trick ...

calohmn commented:
As there has been an update to the Hono chart since you created your branch (see 1b572fb6931f4312949445609bda726eb07f5c6c) you should also do a rebase.

paulocabrita-ionos commented:
Thanks @sophokles73 and @calohmn for your input. 

I will change on my branch the following (and commit):
- {{ .Values.deviceRegistryExample.embeddedJdbcDeviceRegistry.storageSize }}
- in the Chart.yaml file, I will put the version 2.5.1

Do I have to do something more?

Thanks.

calohmn commented:
@paulocabrita-ionos You could also add a short note in the chart `README.md` in the `Release Notes` section about the added feature. That would be all.

Note that the 2 commits you added today have a different email address associated with them (`[number]+[github-username]@users.noreply.github.com` - that's not visible here in the GitHub UI, but I saw it in a local checkout of your branch). For this address, the Eclipse Contributor Agreement check is failing, so these commits need to be changed/overwritten.

I guess it would be easiest, if you delete your local branch, then recreate it based on the current `master` branch HEAD commit. You then make your changes, commit and force-push your branch.

calohmn commented:
@paulocabrita-ionos It seems the change in `values.yaml`
```
    # storageSize contains the size of the storage that will be created using the 
    # storageClass or the cluster's default configuration. 
    # If no value is specified, the default value will be assumed.
    storageSize: 1Mi
```
is missing in your recent [commit](https://github.com/eclipse/packages/compare/46f17aa5c7c89963bfc62fab0ecde62891ef68bc..a7d982487eb3cecc805334034a5984d57d4f41fc).

paulocabrita-ionos commented:
@calohmn I don't know the origin of those lines. In my branch `set_storagesize_config`, the charts/hono/values.yaml does not have those lines. It looks that way:
```
# storageClass indicates the name of the kubernetes StorageClass that
# should be used for the example registry's persistent volume claim.
# If not set, the cluster's default storage class is used.
storageClass:
# cmdLineArgs contains additional arguments to be passed to the container's CMD.
cmdLineArgs: []
# quarkusConfigLocations contains resources that the Quarkus based variant of the component should read
# additional configuration from.
```

I don't know where to look for. Any idea?

calohmn commented:
The origin of those lines is your commit from 2 months ago.
The changes you committed today don't contain them, so it seems you forgot to commit the above mentioned `values.yaml` change in your commit today.

calohmn commented:
(if you click on the "force-pushed" link in the `paulocabrita-ionos force-pushed the set_storagesize_config branch from 46f17aa to a7d9824` notification above, you see a diff between the branch before and after your force-push)

paulocabrita-ionos commented:
Thanks @calohmn . I corrected it.

Just a quick recap for what I done. I changed the following files:
- `hono-service-device-registry-pvc.yaml` adding: storage: {{ .Values.deviceRegistryExample.embeddedJdbcDeviceRegistry.storageSize | quote }};
- `charts/hono/README.md` with the new version `2.5.1`;
- `charts/hono/Chart.yaml` with the new version `2.5.1`;
- `charts/hono/values.yaml` with the default value for storage, `storageSize: 1Mi`;

I tested it without specifying a value for the storageSize and it worked. I also specified a value for the storageSize and it worked like a charm. 

I think that's all.

calohmn commented:
@sophokles73 Could you reassess your request for change?

sophokles73 reviewed (approved):
LGTM

Thanks for contributing :+1:

calohmn commented:
@paulocabrita-ionos Usually, we like to merge PR changes as one commit into the master branch. However, if I squash your commits here, the email-address in the commit will be again from your (currently not eclipse-account-associated) private GitHub address (`[ID]-[yourname]@users.noreply.github.com`). It's because this is your [GitHub commit email address](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-commit-email-addresses), related to having the `Keep my email addresses private` option checked.

If you don't mind, could you choose one of these approaches:
- either associate your private GitHub address with your eclipse account by logging into your eclipse.org account and setting your GitHub Username under "Social Media Links"
- or change your GitHub commit address to the one you used in the commits here (related to `Keep my email addresses private` option)
- or squash commits (or recreate branch with one commit) and force push
